src/editor/components/modals/file-modal/body.jsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import memoize from "memoize-one";
+
+import AddButton from "./add-button";
+import NoFilesNotice from "./no-files-notice";
+import FileList from "./file-list";
+import Footer from "./footer";
+import { fileShape } from "./propShapes";
+import { defaultSpacing } from "./styles";
+
+const bodyPadding = defaultSpacing;
+
+// Padding is not scrollable in Firefox. Wrapping it fixes that.
+// https://stackoverflow.com/a/29987556/4297741
+const BodyWrapper = styled.div`
+  display: grid;
+  height: 100%;
+  overflow: auto;
+`;
+
+const Body = styled.div`
+  align-items: start;
+  display: grid;
+  grid-gap: ${defaultSpacing};
+  grid-template-rows: min-content auto;
+  padding: ${bodyPadding};
+`;
+
+export default class extends React.Component {
+  static propTypes = {
+    files: PropTypes.objectOf(PropTypes.shape(fileShape)).isRequired,
+    onFileSelection: PropTypes.func.isRequired,
+    confirmDelete: PropTypes.func.isRequired
+  };
+
+  hasVisibleFiles = memoize(files => {
```

Comment:
is `memoize-one` necessary here? This function is probably very fast to perform, right? Even in hairy cases (hundreds of files) this function probably wouldn't slow things down if it were run a bunch of times.

Reply:
True, fair enough. I'll remove memoize-one.